src/databricks/labs/ucx/hive_metastore/table_size.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from collections.abc import Iterable
+from dataclasses import dataclass
+from functools import partial
+
+from databricks.labs.ucx.framework.crawlers import CrawlerBase, RuntimeBackend
+from databricks.labs.ucx.hive_metastore import TablesCrawler
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class TableSize:
+    catalog: str
+    database: str
+    name: str
+    size_in_bytes: int
+
+
+class TableSizeCrawler(CrawlerBase):
+    def __init__(self, backend: RuntimeBackend, schema):
+        """
+        Initializes a TablesSizeCrawler instance.
+
+        Args:
+            backend (SqlBackend): The SQL Execution Backend abstraction (either REST API or Spark)
+            schema: The schema name for the inventory persistence.
+        """
+        self._backend: RuntimeBackend = backend
+        super().__init__(backend, "hive_metastore", schema, "table_size", TableSize)
+        self._tables_crawler = TablesCrawler(backend, schema)
+
+    def _crawl(self) -> Iterable[TableSize]:
+        """Crawls and lists tables using table crawler
+        Identifies DBFS root tables and calculates the size for these.
+        """
+        for table in self._tables_crawler.snapshot():
+            if not table.kind == "TABLE":
+                continue
+            if not table.is_dbfs_root():
+                continue
+            size_in_bytes = self._backend.get_table_size(table.key)
```

Comment:
Call spark from here, not from runtime backend.

Reply:
Done